src/select-menu/src/OptionsList.js

```diff
@@ -84,6 +168,24 @@ export default class OptionsList extends PureComponent {
         selected: nextProps.selected
       })
     }
+
+    if (nextProps.options !== this.state.options) {
+      const newCollapsedGroups = getCollapsedGroups(nextProps.options)
+      const collapsedGroups = [
+        ...this.state.collapsedGroups,
+        ...newCollapsedGroups
+      ].reduce((acc, groupLabel) => {
+        if (acc.includes(groupLabel)) return acc
+        return [...acc, groupLabel]
```

Comment:
You shouldn't create a new array on every loop, `acc.push(groupLabel); return acc` instead.

Or use [_.uniq](https://lodash.com/docs/4.17.5#uniq) instead of this reduce.

Reply:
Will take a look at that!